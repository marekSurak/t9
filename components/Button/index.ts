import { styled } from 'styles/stitches.config'

export const Button = styled('button', {
  display: 'flex',
  flexFlow: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  background: '$lightGray',

  border: 'none',
  borderRadius: '50%',
  padding: '$3',
  width: '$5',
  height: '$5',

  '&:hover': {
    background: '$dimGray',
  },
})
