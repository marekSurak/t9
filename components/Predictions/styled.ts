import { styled } from 'styles/stitches.config'

export const PredictionsContainer = styled('ul', {
  display: 'flex',
  gap: '$4',
  alignSelf: 'center',
  overflowX: 'scroll',
  padding: '0 $2 $3',

  // hide scrollbar for Firefox & Chrome
  scrollbarWidth: 'none',
  '&::-webkit-scrollbar': {
    display: 'none',
  },
})

export const PredictionItem = styled('li', {
  fontSize: '$4',

  '&:first-of-type': {
    fontWeight: '$bold',
    fontSize: '$5',
  },
})

export const Separator = styled('div', {
  borderRight: '1px solid $button',
})

export const Loader = styled('div', {
  height: '$2',
  width: '100%',
  borderRadius: '$1',
  background: '$button',
})
