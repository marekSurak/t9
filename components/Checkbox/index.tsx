import type { FC } from 'react'
import React from 'react'

import { CheckboxContainer } from './styled'

interface IProps {
  checked: boolean
  onClick: () => void
}

export const CHECKBOX_ID = 'checkbox'

const Checkbox: FC<IProps> = ({ checked, onClick }) => (
  <CheckboxContainer>
    <label htmlFor={CHECKBOX_ID}>Use word list</label>
    <input
      id={CHECKBOX_ID}
      type="checkbox"
      checked={checked}
      onChange={onClick}
    />
  </CheckboxContainer>
)

export const MemoizedCheckbox = React.memo(Checkbox)
