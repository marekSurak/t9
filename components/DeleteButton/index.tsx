import type { FC } from 'react'

import { Button, ButtonContainer } from './styled'

interface IProps {
  onClick: () => void
  children: React.ReactNode
}

export const DeleteButton: FC<IProps> = ({ onClick, children }) => (
  <ButtonContainer>
    <Button type="button" onClick={onClick}>
      {children}
    </Button>
  </ButtonContainer>
)
