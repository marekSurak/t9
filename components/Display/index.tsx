import type { FC } from 'react'

import { DisplayContainer } from './styled'

interface IProps {
  children: React.ReactNode
}

export const Display: FC<IProps> = ({ children }) => (
  <DisplayContainer>{children}</DisplayContainer>
)
