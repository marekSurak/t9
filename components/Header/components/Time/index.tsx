import type { FC } from 'react'
import { useCallback, useEffect, useState } from 'react'

import { TimeContainer } from './styled'

export const Time: FC = () => {
  const [time, setTime] = useState<string>('')

  const updateTime = useCallback(() => {
    const date = new Date()
    const hours = date.getHours()
    const minutes = date.getMinutes()

    setTime(`${hours}:${minutes}`)
  }, [setTime])

  useEffect(() => {
    updateTime()

    const interval = setInterval(() => updateTime(), 60000)

    return () => {
      clearInterval(interval)
    }
  }, [updateTime])

  return <TimeContainer>{time}</TimeContainer>
}
