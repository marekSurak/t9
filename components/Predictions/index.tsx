import type { FC } from 'react'
import React from 'react'

import type { TPredictions } from 'api/types/predictions'

import { PredictionsContainer, PredictionItem, Separator } from './styled'

interface IProps {
  predictions: TPredictions | undefined
}

export const Predictions: FC<IProps> = ({ predictions }) => {
  if (!predictions || !predictions?.length) return null

  return (
    <PredictionsContainer>
      {predictions.map((prediction, index) => (
        <React.Fragment key={prediction}>
          <PredictionItem>{prediction}</PredictionItem>
          {index < predictions.length - 1 && <Separator />}
        </React.Fragment>
      ))}
    </PredictionsContainer>
  )
}
