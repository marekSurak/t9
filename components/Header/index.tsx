import type { FC } from 'react'
import React from 'react'

import { BatteryIcon } from 'components/icons/BatteryIcon'
import { SignalIcon } from 'components/icons/SignalIcon'
import { WifiIcon } from 'components/icons/WifiIcon'

import { Time } from './components/Time'
import { HeaderContainer, IconsContainer } from './styled'

const Header: FC = () => (
  <HeaderContainer>
    <Time />

    <IconsContainer>
      <SignalIcon />
      <WifiIcon />
      <BatteryIcon />
    </IconsContainer>
  </HeaderContainer>
)

export const MemiozedHeader = React.memo(Header)
