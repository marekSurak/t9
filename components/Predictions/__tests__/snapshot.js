import { render } from '@testing-library/react'
import { Predictions } from 'components/Predictions'
import { predictionsMock } from 'api/mocks/predictions'

it('Renders Predictions unchanged', () => {
  const { container } = render(<Predictions predictions={predictionsMock} />)
  expect(container).toMatchSnapshot()
})
