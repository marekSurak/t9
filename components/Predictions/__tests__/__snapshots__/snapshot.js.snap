// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Predictions unchanged 1`] = `
<div>
  <ul
    class="c-icXWdf"
  >
    <li
      class="c-OQLUz"
    >
      ad
    </li>
    <div
      class="c-fkaaSt"
    />
    <li
      class="c-OQLUz"
    >
      ae
    </li>
    <div
      class="c-fkaaSt"
    />
    <li
      class="c-OQLUz"
    >
      af
    </li>
    <div
      class="c-fkaaSt"
    />
    <li
      class="c-OQLUz"
    >
      bd
    </li>
    <div
      class="c-fkaaSt"
    />
    <li
      class="c-OQLUz"
    >
      be
    </li>
    <div
      class="c-fkaaSt"
    />
    <li
      class="c-OQLUz"
    >
      bf
    </li>
    <div
      class="c-fkaaSt"
    />
    <li
      class="c-OQLUz"
    >
      cd
    </li>
    <div
      class="c-fkaaSt"
    />
    <li
      class="c-OQLUz"
    >
      ce
    </li>
    <div
      class="c-fkaaSt"
    />
    <li
      class="c-OQLUz"
    >
      cf
    </li>
  </ul>
</div>
`;
